content/doc/book/pipeline/syntax.adoc

```diff
@@ -0,0 +1,851 @@
+<1> The `stages` section will typically follow the directives such as `agent`,
+`options`, etc.
+
+==== steps
+
+Defines a series of steps to be executed in a given `stage` directive.
+
+[cols="^10h,>90a",role=syntax]
+|===
+| Required
+| Yes
+
+| Parameters
+| _None_
+
+| Allowed
+| Inside each `stage` block.
+|===
+
+[[steps-example]]
+===== Example
+
+[pipeline]
+----
+// Declarative //
+pipeline {
+    agent any
+    stages {
+        stage('Example') {
+            steps { /* <1> */
+                echo 'Hello World'
+            }
+        }
+    }
+}
+----
+<1> The `steps` section must contain one or more steps.
+
+[[declarative-directives]]
+=== Directives
+
+==== agent
+
+The `agent` directive specifies where the entire Pipeline, or a specific stage,
+will execute in the Jenkins environment depending on where the `agent`
+directive is placed. The directive must be defined at the top-level inside the
+`pipeline` block, but stage-level usage is optional.
+
+
+[cols="^10h,>90a",role=syntax]
+|===
+| Required
+| Yes
+
+| Parameters
+| <<agent-parameters, Described below>>
+
+| Allowed
+| In the top-level `pipeline` block and each `stage` block.
+|===
+
+
+[[agent-parameters]]
+===== Parameters
+
+In order to support the wide variety of use-cases Pipeline authors may have,
+the `agent` directive supports a few different types of parameters. These
+parameters can be applied at the top-level of the `pipeline` block, or within
+each `stage` directive.
+
+any:: Execute the Pipeline, or stage, on any available agent. For example: `agent any`
+none:: When applied at the top-level of the `pipeline` block, each `stage`
+directive will need to contain its own `agent` directive. For example: `agent none`
+label:: Execute the Pipeline, or stage, on the agent available in the Jenkins
+environment with the provided label. For example: `agent { label 'my-defined-label' }`
+docker:: Execute the Pipeline, or stage, with the given container which will be
+dynamically provisioned on a <<../glossary#node, node>> pre-configured to
+accept Docker-based Pipelines, or a node matching the optionally defined
+`label` parameter.  The `docker` also optionally accepts an `args` parameter
+which may contain parameters to pass directly to a `docker run` invocation.
+ For example: `agent { docker: 'maven:3-alpine' }` or
+[source,groovy]
+----
+agent {
+    docker {
+        image: 'maven:3-alpine'
+        label: 'my-defined-label'
+        args:  '-v /tmp:/tmp'
+    }
+}
+----
+
+////
+XXX: https://issues.jenkins-ci.org/browse/JENKINS-41607
+
+dockerfile:: Execute the Pipeline, or stage, with a container built from a
+`Dockerfile` contained in the source repository. Conventionally this can be the
+`Dockerfile` in the root of the source repository: `agent { dockerfile true }`
+
+////
+
+
+
+[[agent-example]]
+===== Example
+
+[pipeline]
+----
+// Declarative //
+pipeline {
+    agent { docker 'maven:3-alpine' } /* <1> */
+    stages {
+        stage('Example Build') {
```

Comment:
In all the examples cited I wanted prefix with "Example" in the stage name so people wouldn't just blindly copy and paste :smile: